src/utils/fetchAPI.js

```diff
@@ -14,8 +13,9 @@ const fetchAPI = async (path, params = {}) => {
     },
   };
 
-  const url = getApiEndpoint() + path;
+  const url = getApiEndpoint(extras.uiUrl) + path;
```

Comment:
In the case of `extras.jwt` we await `getAuthJWT()` in the case we don't get it from `extras`, but for `extras.uiUrl` we don't, so we coudl run into some problems if extras is its default value

Reply:
Can you give me an example? Don't really see how this can happen. In `getApiEndpoint` on line 3 we have:

`const url = new URL(location || window.location.href);`

which reverts to the browser's location if we don't supply one ourselves (i.e. if `extras.uiUrl` is `undefined`).